src/python/k4a/docs/testing.md

```diff
@@ -0,0 +1,119 @@
+# Testing Azure Kinect Python API(K4A)
+
+The tests/ folder contains Python tests to check for the quality of the
+Python K4A library. The tests should be run at least on every code submission
+into the main branch of the repository.
+
+## Test Prerequisites
+
+The following are required in order to run the Python K4A tests.
+
+1. Python (>= 3.6)
+
+2. An internet connection to download required Python packages.
+
+3. The depth engine and k4a libraries are placed in src/python/k4a/src/k4a/_libs.
+   See building.md for more info on these binaries.
+
+4. An attached Azure Kinect device.
+
+## Running tests
+
+### Setting up the test environment
+
+1. Copy the k4a and DE dynamic libraries (.dll or .so) into the folder src/python/k4a/src/k4a/_libs. 
+The file names MUST be k4a.dll and depthengine.dll in Windows, and
+the file names MUST be libk4a.so and libdepthengine.so in Linux.
+
+Note: If the libraries have different names, create a symlink with relative redirection. This may
+      require elevated permissions to create the symlinks.
+
+2. In a command line terminal, create a Python virtual environment and activate it (do not include brackets):
+      cd <repo_root>/src/python/k4a
+      python -m venv <env_name>
+      
+      Windows:
+      .\<env_name>\Scripts\activate
+      
+      Ubuntu:
+      source ./<env_name>/bin/activate
+      
+3. Install the k4a package in development mode. This is required in order to
+   automatically install required Python packages for running the tests, as
+   well as to install the code with the k4a package as the root of the 
+   subpackages and modules. In the command below, make sure to include 
+   ".[test]" with dot and square brackets.
+      pip install -e .[test]
+
+### Running unit tests on the command line
+
+Unit tests execute small pieces of code that run very fast (less than a second).
+These tests do not require any hardware to be attached.
+These tests should be run on every pull request.
+      
+To run the unit tests:
+      
+      python -m pytest tests -k unit
+      
+To capture the results, use an additional --junit-xml=./test_results.xml option.
+
+### Running functional tests on the command line
+
+Functional tests execute code that exercises the functionality of an attached device.
+There are 2 types of functional tests:
+    1. functional_fast - fast functional tests that should be run on every pull request.
+    2. functional - more comprehensive functional tests but these may take a long time.
+      
+To run the functional_fast tests:
+
+      python -m pytest tests -k functional_fast
+      
+To run all functional tests:
+
+      python -m pytest tests -k functional
+      
+To capture the results, use an additional --junit-xml=./test_results.xml option.
+
+### Running performance tests on the command line
```

Comment:
There doesn't seem to be any perf tests right now? Maybe add a comment saying so

Reply:
Ok

Reply:
Added comment about there not being performance tests right now.